kernel/kernel-api/src/test/scala/io/delta/kernel/internal/checksum/ChecksumReadWriteSuite.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.checksum
+
+import io.delta.kernel.data.{ArrayValue, ColumnVector, MapValue, Row}
+import io.delta.kernel.internal.actions.{Format, Metadata, Protocol}
+import io.delta.kernel.internal.data.GenericRow
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.snapshot.SnapshotHint
+import io.delta.kernel.internal.util.InternalUtils.singletonStringColumnVector
+import io.delta.kernel.internal.util.VectorUtils
+import io.delta.kernel.test.{BaseMockJsonHandler, MockEngineUtils}
+import io.delta.kernel.types.{StringType, StructType}
+import io.delta.kernel.utils.CloseableIterator
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.util
+import java.util.{Collections, HashMap, Optional, OptionalLong}
+
+class ChecksumReadWriteSuite extends AnyFunSuite with MockEngineUtils {
```

Comment:
add tests for numFiles or getTableSizeBytes aren't present ==> no checksum file is written

Reply:
we also probably want to test that the entire write does not fail if the checksum write fails, but I think we can probably do this in the PR with the e2e integration

Reply:
also tests for when txnId is not present (let's make it optional as in the protocol)

Reply:
Done for numFiles or getTableSizeBytes aren't present ==> no checksum file is written and when txnId is not present.

for the second one I will make sure to add it in e2e tests.